src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -17,7 +22,29 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    private static final int MAX_ANSWER_ELEMENTS = 5;
+    private static final int END_OF_FILE_CODE = 65535;
+
+    public String getUniqueCharacters(String fileName) throws IOException {
+        Set<Character> sortAnswer = new TreeSet<>();
+        try (FileReader fileReader = new FileReader(fileName)) {
+            char symbol = (char)fileReader.read();
+            while (symbol != END_OF_FILE_CODE) {
+                if (Character.isLetter(symbol)) {
+                    sortAnswer.add(Character.toLowerCase(symbol));
+                }
+                symbol = (char)fileReader.read();
+            }
+        }
+        int counter = 0;
+        StringBuilder answer = new StringBuilder();
+        for (Character element : sortAnswer) {
+            answer.append(element);
+            counter++;
```

Comment:
the for-each loop was created to remove "count++" - like constructions. So, think about using for or while loop

Reply:
An unnecessary variable, I agree. But it is impossible to solve through for, since we have a char iterator of type. Done)